java/searching/binary_search.java

```diff
@@ -0,0 +1,40 @@
+public class binary_search{
+
+    static int binarySearch(int[] arr1, int key){
+    int start = 0;
+    int end = arr1.length - 1;
+
+    while (start <= end) {
+        int middle = (int)Math.floor((start + end) / 2);
+        System.out.println("middle: "+middle);
```

Comment:
Is this required?

Reply:
Yes, actually Math.floor() returns a double value to convert it into int.. requires conversion...